pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test: TestSQLStatsCompactor failed

pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test.TestSQLStatsCompactor [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9923748?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9923748?buildTab=artifacts#/) on master @ [e3efa8895e51c2f018be04c978d8190060e95afa](https://github.com/cockroachdb/cockroach/commits/e3efa8895e51c2f018be04c978d8190060e95afa):


```
=== RUN   TestSQLStatsCompactor
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/9a30d5271a65be6b3f2aa1433be9011f/logTestSQLStatsCompactor1508140712
    test_log_scope.go:79: use -show-logs to present logs inline
=== CONT  TestSQLStatsCompactor
    compaction_test.go:264: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/9a30d5271a65be6b3f2aa1433be9011f/logTestSQLStatsCompactor1508140712
--- FAIL: TestSQLStatsCompactor (127.83s)
=== RUN   TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=205/rowsDeletePerTxn=0
    compaction_test.go:231: 
        	Error Trace:	pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test_test/pkg/sql/sqlstats/persistedsqlstats/compaction_test.go:231
        	Error:      	Not equal: 
        	            	expected: 26
        	            	actual  : 23
        	Test:       	TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=205/rowsDeletePerTxn=0
        	Messages:   	expected 26 number of wide scans issued, but 23 number of wide scan issued
    --- FAIL: TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=205/rowsDeletePerTxn=0 (15.80s)
=== RUN   TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=40/rowsDeletePerTxn=1024
    compaction_test.go:231: 
        	Error Trace:	pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test_test/pkg/sql/sqlstats/persistedsqlstats/compaction_test.go:231
        	Error:      	Not equal: 
        	            	expected: 32
        	            	actual  : 27
        	Test:       	TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=40/rowsDeletePerTxn=1024
        	Messages:   	expected 32 number of wide scans issued, but 27 number of wide scan issued
    --- FAIL: TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=40/rowsDeletePerTxn=1024 (19.03s)
=== RUN   TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=40/rowsDeletePerTxn=2
    compaction_test.go:231: 
        	Error Trace:	pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test_test/pkg/sql/sqlstats/persistedsqlstats/compaction_test.go:231
        	Error:      	Not equal: 
        	            	expected: 32
        	            	actual  : 29
        	Test:       	TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=40/rowsDeletePerTxn=2
        	Messages:   	expected 32 number of wide scans issued, but 29 number of wide scan issued
    --- FAIL: TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=40/rowsDeletePerTxn=2 (30.23s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #102733 pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test: TestSQLStatsCompactor failed [C-test-failure O-robot T-cluster-observability branch-release-23.1]
- #101674 pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test: TestSQLStatsCompactor failed [C-test-failure O-robot T-cluster-observability branch-release-23.1.0]
</p>
</details>
/cc @cockroachdb/cluster-observability
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSQLStatsCompactor.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-27633

pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test.TestSQLStatsCompactor [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9923753?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9923753?buildTab=artifacts#/) on master @ [e3efa8895e51c2f018be04c978d8190060e95afa](https://github.com/cockroachdb/cockroach/commits/e3efa8895e51c2f018be04c978d8190060e95afa):


```
=== RUN   TestSQLStatsCompactor
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/9a30d5271a65be6b3f2aa1433be9011f/logTestSQLStatsCompactor1462361162
    test_log_scope.go:79: use -show-logs to present logs inline
=== CONT  TestSQLStatsCompactor
    compaction_test.go:264: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/9a30d5271a65be6b3f2aa1433be9011f/logTestSQLStatsCompactor1462361162
--- FAIL: TestSQLStatsCompactor (122.95s)
=== RUN   TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=205/rowsDeletePerTxn=0
    compaction_test.go:231: 
        	Error Trace:	pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test_test/pkg/sql/sqlstats/persistedsqlstats/compaction_test.go:231
        	Error:      	Not equal: 
        	            	expected: 27
        	            	actual  : 25
        	Test:       	TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=205/rowsDeletePerTxn=0
        	Messages:   	expected 27 number of wide scans issued, but 25 number of wide scan issued
    --- FAIL: TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=205/rowsDeletePerTxn=0 (17.08s)
=== RUN   TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=40/rowsDeletePerTxn=1024
    compaction_test.go:231: 
        	Error Trace:	pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test_test/pkg/sql/sqlstats/persistedsqlstats/compaction_test.go:231
        	Error:      	Not equal: 
        	            	expected: 32
        	            	actual  : 28
        	Test:       	TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=40/rowsDeletePerTxn=1024
        	Messages:   	expected 32 number of wide scans issued, but 28 number of wide scan issued
    --- FAIL: TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=40/rowsDeletePerTxn=1024 (21.14s)
```
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #102733 pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test: TestSQLStatsCompactor failed [C-test-failure O-robot T-cluster-observability branch-release-23.1]
- #101674 pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test: TestSQLStatsCompactor failed [C-test-failure O-robot T-cluster-observability branch-release-23.1.0]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSQLStatsCompactor.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test.TestSQLStatsCompactor [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9936079?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9936079?buildTab=artifacts#/) on master @ [5c9746012cb9a3244f47098254e603b7d924fdb2](https://github.com/cockroachdb/cockroach/commits/5c9746012cb9a3244f47098254e603b7d924fdb2):


```
=== RUN   TestSQLStatsCompactor
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/9a30d5271a65be6b3f2aa1433be9011f/logTestSQLStatsCompactor1943965938
    test_log_scope.go:79: use -show-logs to present logs inline
=== CONT  TestSQLStatsCompactor
    compaction_test.go:264: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/9a30d5271a65be6b3f2aa1433be9011f/logTestSQLStatsCompactor1943965938
--- FAIL: TestSQLStatsCompactor (148.49s)
=== RUN   TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=40/rowsDeletePerTxn=1024
    compaction_test.go:231: 
        	Error Trace:	pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test_test/pkg/sql/sqlstats/persistedsqlstats/compaction_test.go:231
        	Error:      	Not equal: 
        	            	expected: 32
        	            	actual  : 31
        	Test:       	TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=40/rowsDeletePerTxn=1024
        	Messages:   	expected 32 number of wide scans issued, but 31 number of wide scan issued
    --- FAIL: TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=40/rowsDeletePerTxn=1024 (20.35s)
=== RUN   TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=40/rowsDeletePerTxn=2
    compaction_test.go:231: 
        	Error Trace:	pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test_test/pkg/sql/sqlstats/persistedsqlstats/compaction_test.go:231
        	Error:      	Not equal: 
        	            	expected: 32
        	            	actual  : 31
        	Test:       	TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=40/rowsDeletePerTxn=2
        	Messages:   	expected 32 number of wide scans issued, but 31 number of wide scan issued
    --- FAIL: TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=40/rowsDeletePerTxn=2 (41.55s)
=== RUN   TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=205/rowsDeletePerTxn=0
    compaction_test.go:231: 
        	Error Trace:	pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test_test/pkg/sql/sqlstats/persistedsqlstats/compaction_test.go:231
        	Error:      	Not equal: 
        	            	expected: 24
        	            	actual  : 23
        	Test:       	TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=205/rowsDeletePerTxn=0
        	Messages:   	expected 24 number of wide scans issued, but 23 number of wide scan issued
    --- FAIL: TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=205/rowsDeletePerTxn=0 (17.26s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #102733 pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test: TestSQLStatsCompactor failed [C-test-failure O-robot T-cluster-observability branch-release-23.1]
- #101674 pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test: TestSQLStatsCompactor failed [C-test-failure O-robot T-cluster-observability branch-release-23.1.0]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSQLStatsCompactor.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test.TestSQLStatsCompactor [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9936086?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9936086?buildTab=artifacts#/) on master @ [5c9746012cb9a3244f47098254e603b7d924fdb2](https://github.com/cockroachdb/cockroach/commits/5c9746012cb9a3244f47098254e603b7d924fdb2):


```
=== RUN   TestSQLStatsCompactor
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/9a30d5271a65be6b3f2aa1433be9011f/logTestSQLStatsCompactor118209665
    test_log_scope.go:79: use -show-logs to present logs inline
=== CONT  TestSQLStatsCompactor
    compaction_test.go:264: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/9a30d5271a65be6b3f2aa1433be9011f/logTestSQLStatsCompactor118209665
--- FAIL: TestSQLStatsCompactor (114.63s)
=== RUN   TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=205/rowsDeletePerTxn=0
    compaction_test.go:231: 
        	Error Trace:	pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test_test/pkg/sql/sqlstats/persistedsqlstats/compaction_test.go:231
        	Error:      	Not equal: 
        	            	expected: 27
        	            	actual  : 25
        	Test:       	TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=205/rowsDeletePerTxn=0
        	Messages:   	expected 27 number of wide scans issued, but 25 number of wide scan issued
    --- FAIL: TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=205/rowsDeletePerTxn=0 (16.84s)
=== RUN   TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=40/rowsDeletePerTxn=1024
    compaction_test.go:231: 
        	Error Trace:	pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test_test/pkg/sql/sqlstats/persistedsqlstats/compaction_test.go:231
        	Error:      	Not equal: 
        	            	expected: 32
        	            	actual  : 30
        	Test:       	TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=40/rowsDeletePerTxn=1024
        	Messages:   	expected 32 number of wide scans issued, but 30 number of wide scan issued
    --- FAIL: TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=40/rowsDeletePerTxn=1024 (22.57s)
```
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #102733 pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test: TestSQLStatsCompactor failed [C-test-failure O-robot T-cluster-observability branch-release-23.1]
- #101674 pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test: TestSQLStatsCompactor failed [C-test-failure O-robot T-cluster-observability branch-release-23.1.0]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSQLStatsCompactor.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test.TestSQLStatsCompactor [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9948344?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9948344?buildTab=artifacts#/) on master @ [5ce322985fa93676d5f4c397f609505cb90df9e0](https://github.com/cockroachdb/cockroach/commits/5ce322985fa93676d5f4c397f609505cb90df9e0):


```
=== RUN   TestSQLStatsCompactor
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/9a30d5271a65be6b3f2aa1433be9011f/logTestSQLStatsCompactor1579782962
    test_log_scope.go:79: use -show-logs to present logs inline
=== CONT  TestSQLStatsCompactor
    compaction_test.go:264: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/9a30d5271a65be6b3f2aa1433be9011f/logTestSQLStatsCompactor1579782962
--- FAIL: TestSQLStatsCompactor (145.08s)
=== RUN   TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=205/rowsDeletePerTxn=0
    compaction_test.go:231: 
        	Error Trace:	pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test_test/pkg/sql/sqlstats/persistedsqlstats/compaction_test.go:231
        	Error:      	Not equal: 
        	            	expected: 25
        	            	actual  : 22
        	Test:       	TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=205/rowsDeletePerTxn=0
        	Messages:   	expected 25 number of wide scans issued, but 22 number of wide scan issued
    --- FAIL: TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=205/rowsDeletePerTxn=0 (16.85s)
=== RUN   TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=40/rowsDeletePerTxn=1024
    compaction_test.go:231: 
        	Error Trace:	pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test_test/pkg/sql/sqlstats/persistedsqlstats/compaction_test.go:231
        	Error:      	Not equal: 
        	            	expected: 32
        	            	actual  : 27
        	Test:       	TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=40/rowsDeletePerTxn=1024
        	Messages:   	expected 32 number of wide scans issued, but 27 number of wide scan issued
    --- FAIL: TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=40/rowsDeletePerTxn=1024 (19.63s)
=== RUN   TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=40/rowsDeletePerTxn=2
    compaction_test.go:231: 
        	Error Trace:	pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test_test/pkg/sql/sqlstats/persistedsqlstats/compaction_test.go:231
        	Error:      	Not equal: 
        	            	expected: 32
        	            	actual  : 29
        	Test:       	TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=40/rowsDeletePerTxn=2
        	Messages:   	expected 32 number of wide scans issued, but 29 number of wide scan issued
    --- FAIL: TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=40/rowsDeletePerTxn=2 (38.92s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #102733 pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test: TestSQLStatsCompactor failed [C-test-failure O-robot T-cluster-observability branch-release-23.1]
- #101674 pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test: TestSQLStatsCompactor failed [C-test-failure O-robot T-cluster-observability branch-release-23.1.0]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSQLStatsCompactor.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test.TestSQLStatsCompactor [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9948347?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9948347?buildTab=artifacts#/) on master @ [5ce322985fa93676d5f4c397f609505cb90df9e0](https://github.com/cockroachdb/cockroach/commits/5ce322985fa93676d5f4c397f609505cb90df9e0):


```
=== RUN   TestSQLStatsCompactor
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/9a30d5271a65be6b3f2aa1433be9011f/logTestSQLStatsCompactor267388034
    test_log_scope.go:79: use -show-logs to present logs inline
=== CONT  TestSQLStatsCompactor
    compaction_test.go:264: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/9a30d5271a65be6b3f2aa1433be9011f/logTestSQLStatsCompactor267388034
--- FAIL: TestSQLStatsCompactor (102.77s)
=== RUN   TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=205/rowsDeletePerTxn=0
    compaction_test.go:231: 
        	Error Trace:	pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test_test/pkg/sql/sqlstats/persistedsqlstats/compaction_test.go:231
        	Error:      	Not equal: 
        	            	expected: 27
        	            	actual  : 23
        	Test:       	TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=205/rowsDeletePerTxn=0
        	Messages:   	expected 27 number of wide scans issued, but 23 number of wide scan issued
    --- FAIL: TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=205/rowsDeletePerTxn=0 (16.05s)
```
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #102733 pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test: TestSQLStatsCompactor failed [C-test-failure O-robot T-cluster-observability branch-release-23.1]
- #101674 pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test: TestSQLStatsCompactor failed [C-test-failure O-robot T-cluster-observability branch-release-23.1.0]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSQLStatsCompactor.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test.TestSQLStatsCompactor [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9970284?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9970284?buildTab=artifacts#/) on master @ [53abb2f174283b3de637774e3dbbaeeaa23cde65](https://github.com/cockroachdb/cockroach/commits/53abb2f174283b3de637774e3dbbaeeaa23cde65):


```
=== RUN   TestSQLStatsCompactor
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/9a30d5271a65be6b3f2aa1433be9011f/logTestSQLStatsCompactor731368778
    test_log_scope.go:79: use -show-logs to present logs inline
=== CONT  TestSQLStatsCompactor
    compaction_test.go:264: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/9a30d5271a65be6b3f2aa1433be9011f/logTestSQLStatsCompactor731368778
--- FAIL: TestSQLStatsCompactor (145.62s)
=== RUN   TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=40/rowsDeletePerTxn=1024
    compaction_test.go:231: 
        	Error Trace:	pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test_test/pkg/sql/sqlstats/persistedsqlstats/compaction_test.go:231
        	Error:      	Not equal: 
        	            	expected: 32
        	            	actual  : 29
        	Test:       	TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=40/rowsDeletePerTxn=1024
        	Messages:   	expected 32 number of wide scans issued, but 29 number of wide scan issued
    --- FAIL: TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=40/rowsDeletePerTxn=1024 (20.75s)
=== RUN   TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=40/rowsDeletePerTxn=2
    compaction_test.go:231: 
        	Error Trace:	pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test_test/pkg/sql/sqlstats/persistedsqlstats/compaction_test.go:231
        	Error:      	Not equal: 
        	            	expected: 32
        	            	actual  : 30
        	Test:       	TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=40/rowsDeletePerTxn=2
        	Messages:   	expected 32 number of wide scans issued, but 30 number of wide scan issued
    --- FAIL: TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=40/rowsDeletePerTxn=2 (40.28s)
=== RUN   TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=205/rowsDeletePerTxn=0
    compaction_test.go:231: 
        	Error Trace:	pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test_test/pkg/sql/sqlstats/persistedsqlstats/compaction_test.go:231
        	Error:      	Not equal: 
        	            	expected: 25
        	            	actual  : 23
        	Test:       	TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=205/rowsDeletePerTxn=0
        	Messages:   	expected 25 number of wide scans issued, but 23 number of wide scan issued
    --- FAIL: TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=205/rowsDeletePerTxn=0 (16.72s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #102733 pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test: TestSQLStatsCompactor failed [C-test-failure O-robot T-cluster-observability branch-release-23.1]
- #101674 pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test: TestSQLStatsCompactor failed [C-test-failure O-robot T-cluster-observability branch-release-23.1.0]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSQLStatsCompactor.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test.TestSQLStatsCompactor [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9970290?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9970290?buildTab=artifacts#/) on master @ [53abb2f174283b3de637774e3dbbaeeaa23cde65](https://github.com/cockroachdb/cockroach/commits/53abb2f174283b3de637774e3dbbaeeaa23cde65):


```
=== RUN   TestSQLStatsCompactor
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/9a30d5271a65be6b3f2aa1433be9011f/logTestSQLStatsCompactor3456276133
    test_log_scope.go:79: use -show-logs to present logs inline
=== CONT  TestSQLStatsCompactor
    compaction_test.go:264: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/9a30d5271a65be6b3f2aa1433be9011f/logTestSQLStatsCompactor3456276133
--- FAIL: TestSQLStatsCompactor (112.79s)
=== RUN   TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=205/rowsDeletePerTxn=0
    compaction_test.go:231: 
        	Error Trace:	pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test_test/pkg/sql/sqlstats/persistedsqlstats/compaction_test.go:231
        	Error:      	Not equal: 
        	            	expected: 27
        	            	actual  : 24
        	Test:       	TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=205/rowsDeletePerTxn=0
        	Messages:   	expected 27 number of wide scans issued, but 24 number of wide scan issued
    --- FAIL: TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=205/rowsDeletePerTxn=0 (16.51s)
=== RUN   TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=40/rowsDeletePerTxn=1024
    compaction_test.go:231: 
        	Error Trace:	pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test_test/pkg/sql/sqlstats/persistedsqlstats/compaction_test.go:231
        	Error:      	Not equal: 
        	            	expected: 32
        	            	actual  : 28
        	Test:       	TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=40/rowsDeletePerTxn=1024
        	Messages:   	expected 32 number of wide scans issued, but 28 number of wide scan issued
    --- FAIL: TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=40/rowsDeletePerTxn=1024 (22.20s)
```
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #102733 pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test: TestSQLStatsCompactor failed [C-test-failure O-robot T-cluster-observability branch-release-23.1]
- #101674 pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test: TestSQLStatsCompactor failed [C-test-failure O-robot T-cluster-observability branch-release-23.1.0]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSQLStatsCompactor.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test.TestSQLStatsCompactor [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9982006?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9982006?buildTab=artifacts#/) on master @ [e8e5006c1a4e57688b9da28c182208b36a074248](https://github.com/cockroachdb/cockroach/commits/e8e5006c1a4e57688b9da28c182208b36a074248):


```
=== RUN   TestSQLStatsCompactor
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/9a30d5271a65be6b3f2aa1433be9011f/logTestSQLStatsCompactor3257941985
    test_log_scope.go:79: use -show-logs to present logs inline
=== CONT  TestSQLStatsCompactor
    compaction_test.go:264: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/9a30d5271a65be6b3f2aa1433be9011f/logTestSQLStatsCompactor3257941985
--- FAIL: TestSQLStatsCompactor (149.18s)
=== RUN   TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=205/rowsDeletePerTxn=0
    compaction_test.go:231: 
        	Error Trace:	pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test_test/pkg/sql/sqlstats/persistedsqlstats/compaction_test.go:231
        	Error:      	Not equal: 
        	            	expected: 25
        	            	actual  : 22
        	Test:       	TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=205/rowsDeletePerTxn=0
        	Messages:   	expected 25 number of wide scans issued, but 22 number of wide scan issued
    --- FAIL: TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=205/rowsDeletePerTxn=0 (17.66s)
=== RUN   TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=40/rowsDeletePerTxn=1024
    compaction_test.go:231: 
        	Error Trace:	pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test_test/pkg/sql/sqlstats/persistedsqlstats/compaction_test.go:231
        	Error:      	Not equal: 
        	            	expected: 32
        	            	actual  : 28
        	Test:       	TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=40/rowsDeletePerTxn=1024
        	Messages:   	expected 32 number of wide scans issued, but 28 number of wide scan issued
    --- FAIL: TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=40/rowsDeletePerTxn=1024 (19.78s)
=== RUN   TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=40/rowsDeletePerTxn=2
    compaction_test.go:231: 
        	Error Trace:	pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test_test/pkg/sql/sqlstats/persistedsqlstats/compaction_test.go:231
        	Error:      	Not equal: 
        	            	expected: 32
        	            	actual  : 29
        	Test:       	TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=40/rowsDeletePerTxn=2
        	Messages:   	expected 32 number of wide scans issued, but 29 number of wide scan issued
    --- FAIL: TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=40/rowsDeletePerTxn=2 (41.35s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #102733 pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test: TestSQLStatsCompactor failed [C-test-failure O-robot T-cluster-observability branch-release-23.1]
- #101674 pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test: TestSQLStatsCompactor failed [C-test-failure O-robot T-cluster-observability branch-release-23.1.0]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSQLStatsCompactor.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test.TestSQLStatsCompactor [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9997710?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9997710?buildTab=artifacts#/) on master @ [76c651d6718868e57b7891bba8f7366a25c920d8](https://github.com/cockroachdb/cockroach/commits/76c651d6718868e57b7891bba8f7366a25c920d8):


```
=== RUN   TestSQLStatsCompactor
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/9a30d5271a65be6b3f2aa1433be9011f/logTestSQLStatsCompactor341164160
    test_log_scope.go:79: use -show-logs to present logs inline
=== CONT  TestSQLStatsCompactor
    compaction_test.go:264: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/9a30d5271a65be6b3f2aa1433be9011f/logTestSQLStatsCompactor341164160
--- FAIL: TestSQLStatsCompactor (148.15s)
=== RUN   TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=205/rowsDeletePerTxn=0
    compaction_test.go:231: 
        	Error Trace:	pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test_test/pkg/sql/sqlstats/persistedsqlstats/compaction_test.go:231
        	Error:      	Not equal: 
        	            	expected: 25
        	            	actual  : 23
        	Test:       	TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=205/rowsDeletePerTxn=0
        	Messages:   	expected 25 number of wide scans issued, but 23 number of wide scan issued
    --- FAIL: TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=205/rowsDeletePerTxn=0 (17.42s)
=== RUN   TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=40/rowsDeletePerTxn=1024
    compaction_test.go:231: 
        	Error Trace:	pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test_test/pkg/sql/sqlstats/persistedsqlstats/compaction_test.go:231
        	Error:      	Not equal: 
        	            	expected: 32
        	            	actual  : 29
        	Test:       	TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=40/rowsDeletePerTxn=1024
        	Messages:   	expected 32 number of wide scans issued, but 29 number of wide scan issued
    --- FAIL: TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=40/rowsDeletePerTxn=1024 (20.10s)
=== RUN   TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=40/rowsDeletePerTxn=2
    compaction_test.go:231: 
        	Error Trace:	pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test_test/pkg/sql/sqlstats/persistedsqlstats/compaction_test.go:231
        	Error:      	Not equal: 
        	            	expected: 32
        	            	actual  : 30
        	Test:       	TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=40/rowsDeletePerTxn=2
        	Messages:   	expected 32 number of wide scans issued, but 30 number of wide scan issued
    --- FAIL: TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=40/rowsDeletePerTxn=2 (41.10s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #102733 pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test: TestSQLStatsCompactor failed [C-test-failure O-robot T-cluster-observability branch-release-23.1]
- #101674 pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test: TestSQLStatsCompactor failed [C-test-failure O-robot T-cluster-observability branch-release-23.1.0]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSQLStatsCompactor.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test.TestSQLStatsCompactor [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9997717?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9997717?buildTab=artifacts#/) on master @ [76c651d6718868e57b7891bba8f7366a25c920d8](https://github.com/cockroachdb/cockroach/commits/76c651d6718868e57b7891bba8f7366a25c920d8):


```
=== RUN   TestSQLStatsCompactor
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/9a30d5271a65be6b3f2aa1433be9011f/logTestSQLStatsCompactor3660324918
    test_log_scope.go:79: use -show-logs to present logs inline
=== CONT  TestSQLStatsCompactor
    compaction_test.go:264: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/9a30d5271a65be6b3f2aa1433be9011f/logTestSQLStatsCompactor3660324918
--- FAIL: TestSQLStatsCompactor (97.53s)
=== RUN   TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=40/rowsDeletePerTxn=1024
    compaction_test.go:231: 
        	Error Trace:	pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test_test/pkg/sql/sqlstats/persistedsqlstats/compaction_test.go:231
        	Error:      	Not equal: 
        	            	expected: 32
        	            	actual  : 28
        	Test:       	TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=40/rowsDeletePerTxn=1024
        	Messages:   	expected 32 number of wide scans issued, but 28 number of wide scan issued
    --- FAIL: TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=40/rowsDeletePerTxn=1024 (18.94s)
=== RUN   TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=205/rowsDeletePerTxn=0
    compaction_test.go:231: 
        	Error Trace:	pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test_test/pkg/sql/sqlstats/persistedsqlstats/compaction_test.go:231
        	Error:      	Not equal: 
        	            	expected: 27
        	            	actual  : 24
        	Test:       	TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=205/rowsDeletePerTxn=0
        	Messages:   	expected 27 number of wide scans issued, but 24 number of wide scan issued
    --- FAIL: TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=205/rowsDeletePerTxn=0 (17.46s)
```
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #102733 pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test: TestSQLStatsCompactor failed [C-test-failure O-robot T-cluster-observability branch-release-23.1]
- #101674 pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test: TestSQLStatsCompactor failed [C-test-failure O-robot T-cluster-observability branch-release-23.1.0]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSQLStatsCompactor.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test.TestSQLStatsCompactor [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/10014139?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/10014139?buildTab=artifacts#/) on master @ [29523d3549402b7cdced8fe211e6c133822e5d5a](https://github.com/cockroachdb/cockroach/commits/29523d3549402b7cdced8fe211e6c133822e5d5a):


```
=== RUN   TestSQLStatsCompactor
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/9a30d5271a65be6b3f2aa1433be9011f/logTestSQLStatsCompactor2886969227
    test_log_scope.go:79: use -show-logs to present logs inline
=== CONT  TestSQLStatsCompactor
    compaction_test.go:264: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/9a30d5271a65be6b3f2aa1433be9011f/logTestSQLStatsCompactor2886969227
--- FAIL: TestSQLStatsCompactor (85.90s)
=== RUN   TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=205/rowsDeletePerTxn=0
    compaction_test.go:231: 
        	Error Trace:	pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test_test/pkg/sql/sqlstats/persistedsqlstats/compaction_test.go:231
        	Error:      	Not equal: 
        	            	expected: 29
        	            	actual  : 26
        	Test:       	TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=205/rowsDeletePerTxn=0
        	Messages:   	expected 29 number of wide scans issued, but 26 number of wide scan issued
    --- FAIL: TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=205/rowsDeletePerTxn=0 (15.82s)
```
<p>Parameters: <code>TAGS=bazel,gss,race</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #102733 pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test: TestSQLStatsCompactor failed [C-test-failure O-robot T-cluster-observability branch-release-23.1]
- #101674 pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test: TestSQLStatsCompactor failed [C-test-failure O-robot T-cluster-observability branch-release-23.1.0]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSQLStatsCompactor.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test.TestSQLStatsCompactor [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/10032497?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/10032497?buildTab=artifacts#/) on master @ [0c00e43e434e368be430b41cb83588b36359fb3f](https://github.com/cockroachdb/cockroach/commits/0c00e43e434e368be430b41cb83588b36359fb3f):


```
=== RUN   TestSQLStatsCompactor
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/9a30d5271a65be6b3f2aa1433be9011f/logTestSQLStatsCompactor84976264
    test_log_scope.go:79: use -show-logs to present logs inline
=== CONT  TestSQLStatsCompactor
    compaction_test.go:264: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/9a30d5271a65be6b3f2aa1433be9011f/logTestSQLStatsCompactor84976264
--- FAIL: TestSQLStatsCompactor (128.80s)
=== RUN   TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=205/rowsDeletePerTxn=0
    compaction_test.go:231: 
        	Error Trace:	pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test_test/pkg/sql/sqlstats/persistedsqlstats/compaction_test.go:231
        	Error:      	Not equal: 
        	            	expected: 25
        	            	actual  : 24
        	Test:       	TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=205/rowsDeletePerTxn=0
        	Messages:   	expected 25 number of wide scans issued, but 24 number of wide scan issued
    --- FAIL: TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=205/rowsDeletePerTxn=0 (15.63s)
=== RUN   TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=40/rowsDeletePerTxn=1024
    compaction_test.go:231: 
        	Error Trace:	pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test_test/pkg/sql/sqlstats/persistedsqlstats/compaction_test.go:231
        	Error:      	Not equal: 
        	            	expected: 32
        	            	actual  : 31
        	Test:       	TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=40/rowsDeletePerTxn=1024
        	Messages:   	expected 32 number of wide scans issued, but 31 number of wide scan issued
    --- FAIL: TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=40/rowsDeletePerTxn=1024 (22.84s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #102733 pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test: TestSQLStatsCompactor failed [C-test-failure O-robot T-cluster-observability branch-release-23.1]
- #101674 pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test: TestSQLStatsCompactor failed [C-test-failure O-robot T-cluster-observability branch-release-23.1.0]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSQLStatsCompactor.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/sql/sqlstats/persistedsqlstats/compaction_test.go
+++ b/pkg/sql/sqlstats/persistedsqlstats/compaction_test.go
@@ -33,7 +33,6 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/sql/sqlstats/persistedsqlstats"
 	"github.com/cockroachdb/cockroach/pkg/sql/tests"
 	"github.com/cockroachdb/cockroach/pkg/testutils/serverutils"
-	"github.com/cockroachdb/cockroach/pkg/testutils/skip"
 	"github.com/cockroachdb/cockroach/pkg/testutils/sqlutils"
 	"github.com/cockroachdb/cockroach/pkg/util/encoding"
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
@@ -69,7 +68,6 @@ func TestSQLStatsCompactorNilTestingKnobCheck(t *testing.T) {
 func TestSQLStatsCompactor(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
-	skip.WithIssue(t, 102750)
 	ctx := context.Background()
 
 	testCases := []struct {
@@ -118,44 +116,45 @@ func TestSQLStatsCompactor(t *testing.T) {
 		},
 	}
 
-	kvInterceptor := kvScanInterceptor{}
-	cleanupInterceptor := cleanupInterceptor{}
-
-	server, conn, _ := serverutils.StartServer(
-		t, base.TestServerArgs{
-			Knobs: base.TestingKnobs{
-				SQLStatsKnobs: &sqlstats.TestingKnobs{
-					AOSTClause: "AS OF SYSTEM TIME '-1us'",
-					StubTimeNow: func() time.Time {
-						return timeutil.Now().Add(-2 * time.Hour)
-					},
-				},
-				Store: &kvserver.StoreTestingKnobs{
-					TestingRequestFilter: kvInterceptor.intercept,
-				},
-			},
-		},
-	)
-
-	defer server.Stopper().Stop(ctx)
-
-	sqlConn := sqlutils.MakeSQLRunner(conn)
-	internalExecutor := server.InternalExecutor().(isql.Executor)
-
-	// Disable automatic flush since the test will handle the flush manually.
-	sqlConn.Exec(t, "SET CLUSTER SETTING sql.stats.flush.interval = '24h'")
-	// Disable activity update flush which also does a scan on the stats table
-	sqlConn.Exec(t, "SET CLUSTER SETTING sql.stats.activity.flush.enabled = false")
-	// Change the automatic compaction job to avoid it running during the test.
-	// Test creates a new compactor and calls it directly.
-	sqlConn.Exec(t, "SET CLUSTER SETTING sql.stats.cleanup.recurrence = '@yearly';")
-
 	for _, tc := range testCases {
 		t.Run(fmt.Sprintf("stmtCount=%d/maxPersistedRowLimit=%d/rowsDeletePerTxn=%d",
 			tc.stmtCount,
 			tc.maxPersistedRowLimit,
 			tc.rowsToDeletePerTxn,
 		), func(t *testing.T) {
+
+			kvInterceptor := kvScanInterceptor{}
+			cleanupInterceptor := cleanupInterceptor{}
+
+			server, conn, _ := serverutils.StartServer(
+				t, base.TestServerArgs{
+					Knobs: base.TestingKnobs{
+						SQLStatsKnobs: &sqlstats.TestingKnobs{
+							AOSTClause: "AS OF SYSTEM TIME '-1us'",
+							StubTimeNow: func() time.Time {
+								return timeutil.Now().Add(-2 * time.Hour)
+							},
+						},
+						Store: &kvserver.StoreTestingKnobs{
+							TestingRequestFilter: kvInterceptor.intercept,
+						},
+					},
+				},
+			)
+
+			defer server.Stopper().Stop(ctx)
+
+			sqlConn := sqlutils.MakeSQLRunner(conn)
+			internalExecutor := server.InternalExecutor().(isql.Executor)
+
+			// Disable automatic flush since the test will handle the flush manually.
+			sqlConn.Exec(t, "SET CLUSTER SETTING sql.stats.flush.interval = '24h'")
+			// Disable activity update flush which also does a scan on the stats table
+			sqlConn.Exec(t, "SET CLUSTER SETTING sql.stats.activity.flush.enabled = false")
+			// Change the automatic compaction job to avoid it running during the test.
+			// Test creates a new compactor and calls it directly.
+			sqlConn.Exec(t, "SET CLUSTER SETTING sql.stats.cleanup.recurrence = '@yearly';")
+
 			_, err := internalExecutor.ExecEx(
 				ctx,
 				"truncate-stmt-stats",

